clusterwithcoherence/README.md

```diff
@@ -0,0 +1,119 @@
+* For "Credentails for Virtual Machines and WebLogic"
+   * For the "Password for admin account of VMs", enter 'Secret123456'. 
+   * For the "Password for WebLogic Administrator", enter 'Secret123456'.
+* For Number of VMs, change the value to 3.
+* For "Optional Basic Configuration", ensure  `Yes` is selected to accept default for optional configuration.
+* Click Next.
+* In "TLS/SSL Configuration", leave "Configure WebLogic Administration Console on HTTPS (Secure) port, with your own TLS/SSL Certificate?" with `No` selected.
+* In the "Azure Application Gateway" use these values
+   * Toggle "Connect to Azure Application Gateway" to `Yes`.
+   * Choose "Generate a self-signed certificate" for the "Select desired TLS/SSL certificate option".
+   * To auto-generate a self-signed certificate, you will need to add user-assigned managed identities(at least one), click "Add".
+   * In the window pops from right, choose weblogic-cafe-managed-identity-`<your suffix>`, then click "Add"
+* Click Next.
+* In "Database" use these values
+   * Toggle "Connect to DataBase" to `Yes`. 
+   * For "Choose database type", from the dropdown menu, select the option for "Azure Database for PostgreSQL".
+   * Specify JNDI Name to be 'jdbc/WebLogicCafeDB'. 
+   * Specify DataSource Connection String to be 'jdbc:postgresql://weblogic-cafe-db-`<your suffix>`.postgres.database.azure.com:5432/postgres'
+   * Specify the Database Username to be 'postgres@weblogic-cafe-db-`<your suffix>`'
+   * Enter the Database Password as 'Secret123!' 
+* Click Next.
+* In "DNS Configuration", leave "Configure Custom DNS Alias?" with `No` selected.
+* Click Next.
+* In "Azure Active Directory", leave "Connect to Active Directory" with `No` selected.
+* Click Next.
+* In "Elasticsearch and Kibana", leave "Export logs to Elasticsearch?" with `No` selected.
+* Click Next.
+* In "Coherence", toggle "Use Coherence cache?" to `Yes`. Leave other values to default.
+* Click Next:Review + create
+   * On the Summary blade you must see "Validation passed".  If you don't see this, you must troubleshoot and resolve the reason.  After you have done so, you can continue.
+   * On the final screen, click Create.
+* It will take some time for the WebLogic cluster to properly deploy (could be up to an hour). Once the deployment completes, in the portal go to 'All resources'.
+* Find and click on adminVM. Copy the DNS name for the admin server. You should be able to log onto http://`<admin server DNS name>`:7001/console successfully using the credentials above.  If you are not able to log in, you must troubleshoot and resolve the reason why before continuing.
+
+## Prepare Coherence App
+* Open a command line window and cd to `<your path to the repo>/weblogic-on-azure/clusterwithcoherence/coherence/gar`.
+* Run "jar cvf ../ear/weblogic-cafe-coherence.gar *", this will generate a weblogic-cafe-coherence.gar file to `<your path to the repo>/weblogic-on-azure/clusterwithcoherence/coherence/ear`.
```

Comment:
Run "jar cvf ../ear/weblogic-cafe-coherence.gar *" -> Run `jar cvf ../ear/weblogic-cafe-coherence.gar *`

Reply:
Can we use `mvn build` to generate the `gar` file?

Reply:
Please refer to [Building a Real Application with Maven](https://docs.oracle.com/middleware/1221/core/MAVEN/real_app.htm#MAVEN8918)